platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r920/ResilientGradleBuildSyncCrossVersionSpec.groovy

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r920
+
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.tooling.BuildAction
+import org.gradle.tooling.BuildController
+import org.gradle.tooling.model.ProjectIdentifier
+import org.gradle.tooling.model.gradle.ResilientGradleBuild
+import org.gradle.tooling.model.kotlin.dsl.KotlinDslScriptModel
+
+@ToolingApiVersion('>=9.2')
+@TargetGradleVersion('>=9.2')
+class ResilientGradleBuildSyncCrossVersionSpec extends ToolingApiSpecification {
+
+    def setup() {
+        settingsFile.delete() // This is automatically created by `ToolingApiSpecification`
+    }
+
+    def runCustomModelAction() {
+        succeeds {
+            action(new CustomModelAction())
+                .withArguments("-Dorg.gradle.internal.resilient-model-building=true")
+                .run()
+        }
+    }
+
+    def "broken settings file - strict mode - build action"() {
```

Comment:
Please clean up test names a bit. 

I think "strict mode" should never be there, because we gave up on using the "lenient mode", so no need to specify "strict mode" explicitly.

Also, we always use build actions, so no need to specify "build action" in test method names either.